### ISSUE TITLE
Found `Failure/Error: invalid` when stdin uses a json previous result

### ISSUE THREAD

**Issue by developer-kikikaikai** [k1LoW/runn#520]
## Preparation

Use the yaml

```senario.yaml
desc: Exec test
steps:
  -
    exec:
      command: echo -n ‘{“field”:“hello world!!“}’
  -
    exec:
      command: cat
      stdin: ‘{{ steps[0].stdout }}’
```

## Result

The stdin is failed to parse.

```
F

1) test.yaml
 Failure/Error: invalid ‘Exec test’.steps[1]: map[command:cat stdin:map[field:hello world!!]]

1 scenario, 0 skipped, 1 failure
exit status 1
```

It  should be parsed.

## Envionment
I found the issue on debian,
And it reproduces on runn docker.

```bash
#copy senario
docker pull ghcr.io/k1low/runn:latest
docker run --rm --name runn -v /path/to/your/senario/dir:/books ghcr.io/k1low/runn:latest run /books/test.yaml
```

**Comment by developer-kikikaikai** [comment]
I cloned runn repository, and run the scenario with debug print code.

It seems that [ReplaceYAML](https://github.com/k1LoW/runn/blob/3472f071c3014589dbbb39cd332a2cefce2cb18e/eval.go#LL88C1-L89C1) returns json format string as following.

```
command: cat
stdin: {"field":"hello world!!"}
```

But [yaml.Unmarshall](https://github.com/k1LoW/runn/blob/3472f071c3014589dbbb39cd332a2cefce2cb18e/eval.go#L93) returns map.
```
map[command:cat stdin:map[field:hello world!!]]
```

**Comment by k1LoW** [comment]
@developer-kikikaikai Thank you for your report!

**Comment by k1LoW** [comment]
It is a runn specification that values that can be recognized as JSON are expanded to map and slice ( e.g. HTTP Request/Response and JSONL files).

However, we recognize that it is a problem if you want to pass it as a string but cannot.

**Comment by k1LoW** [comment]
> If it's a runn specification, I'd like to know how to get the field value of json field from the 1st step.
> In other words, I'd like to know how we recognize json value with "command".

You're right, there's no way to get the value of a field in the expanded JSON, different from the other cases. Thank you.

u-m...

**Comment by k1LoW** [comment]
Hi @developer-kikikaikai 

We Added built-in function to manipulate JSON in v0.75.3.

```yaml
desc: Exec test
steps:
  -
    exec:
      command: echo -n '{"field":"hello world!!"}'
  -
    exec:
      command: cat
      stdin: '{{ json.Decode(steps[0].stdout).field }}'
    test: current.stdout == 'hello world!!'
```

**Comment by developer-kikikaikai** [comment]
Thanks, it's solved! Thank you for your support!
